### PR TITLE
Check selected element's validity before device-to-host transfer

### PR DESCRIPTION
**Pull request by jakirkham**
As the device-to-host transfer here can be a bit slow, make sure to check that it is needed (and skip it if not) before performing it.

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -414,14 +414,17 @@ def element_indexing(self, index):
         if val.null_count == 1:
             val = None
         else:
-            val = val.to_array()[0]
+            valid = (
+                cudautils.mask_get.py_func(self.mask_array_view, index)
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/column.py:418]: This will add a kernel invocation and a device to host. I don't think we want to do this.

Comment (jakirkham) [inline python/cudf/cudf/core/column/column.py:418]: Maybe I'm missing something, but does the selection/`null_count` check above already handle this case?

Comment (jakirkham) [reply python/cudf/cudf/core/column/column.py:418]: It was already happening before though.

However I'm now wondering it this is actually needed. Would the selection/`null_count` check above already handle this case?

Comment (jakirkham) [reply python/cudf/cudf/core/column/column.py:418]: Based on offline discussion, it sounds like this isn't needed. Dropping in PR ( https://github.com/rapidsai/cudf/pull/4912 ) and closing this PR out.

**Pull request closed without merging**